src/services/DispatchService.ts

```diff
@@ -0,0 +1,150 @@
+        console.log("eventPayload: ", eventPayload);
+
+        const eventType = eventPayload.eventType; //INSERT, MODIFY or REMOVE
+
+        switch (eventType) {
+            case EVENT_TYPE.INSERT:
+                return this.sendPost(eventPayload,target);
+            case EVENT_TYPE.MODIFY:
+                return this.sendPut(eventPayload, target);
+            case EVENT_TYPE.REMOVE:
+                return this.sendDelete(eventPayload, target);
+            default:
+                console.error(ERROR.NO_HANDLER_METHOD);
+                return this.sendRecordToDLQ(eventPayload, target);
+        }
+    }
+
+    private async sendPost(event: IBody, target: ITarget): Promise<any> {
+        const updateContent = DynamoDB.Converter.unmarshall(event.body.NewImage);
+        const path = this.processPath(target.endpoints.INSERT, event.body);
+        if(await this.isValidMessageBody(updateContent, target)) {
+            try {
+                return await this.dao.postMessage(updateContent, path);
+            } catch (e) {
+                console.log("Failed to send request: ", e);
+                if (this.isRetryableError(e)) {
+                    return Promise.reject(ERROR.FAILED_TO_SEND);
+                } else {
+                    return this.sendRecordToDLQ(event, target);
+                }
+            }
+        } else {
+            console.log(ERROR.FAILED_VALIDATION_SENDING_TO_DLQ);
+            return this.sendRecordToDLQ(event, target);
+        }
+    }
+
+    private async sendPut(event: IBody, target: ITarget): Promise<any> {
+        const updateContent = DynamoDB.Converter.unmarshall(event.body.NewImage);
+        const path = this.processPath(target.endpoints.MODIFY, event.body);
+        if(await this.isValidMessageBody(updateContent, target)) {
+            try {
+                return await this.dao.putMessage(updateContent, path);
+            } catch (e) {
+                console.log("Failed to send request: ", e);
+                if (this.isRetryableError(e)) {
+                    return Promise.reject(ERROR.FAILED_TO_SEND);
+                } else {
+                    return this.sendRecordToDLQ(event, target);
+                }
+            }
+        } else {
+            console.log(ERROR.FAILED_VALIDATION_SENDING_TO_DLQ);
+            return this.sendRecordToDLQ(event, target);
+        }
+    }
+
+    private async sendDelete(event: IBody, target: ITarget): Promise<any> {
+        const path = this.processPath(target.endpoints.REMOVE, event.body);
+        try {
+            return await this.dao.deleteMessage(path);
+        } catch (e) {
+            console.log("Failed to send request: ", e);
+            if (this.isRetryableError(e)) {
+                return Promise.reject(ERROR.FAILED_TO_SEND);
+            } else {
+                return this.sendRecordToDLQ(event, target);
+            }
+        }
+    }
+
+    public processPath(path: string, body: any) {
```

Comment:
What does this function do?

Reply:
Ah, I see the expected behaviour in the unit tests. Wouldn't it be easier to extract the testResultId from the parsed JSON ?

Reply:
I don't understand the suggestion here. We need a generic way of retrieving the key value, which differs from data type to data type. The DB Stream produces a separate list of keys for easy access.

Reply:
Ah ok, understood. I was just thinking if there was a way to avoid so much regEx manipulation on a JSON file.

Reply:
paths come from config. It's the same regex replacement from config.